src/workflows/airqo_etl_utils/ml_utils.py

```
@@ -741,3 +765,111 @@
+    @staticmethod
+    def encode(data: pd.DataFrame, encoder: str = "LabelEncoder") -> pd.DataFrame:
+        """
+        applies encoding for the city and country features
+
+        Keyword arguments:
+        data --  the data frame to apply the transformation on
+        encoder --  the type of encoding to apply (default: 'LabelEncoder')
+        Return: returns a dataframe after applying the encoding
+        """
+
+        if not "city" in data.columns:
```

Ruff

Test for membership should be `not in`

Convert to `not in`

(E713)